src/spyglass/spikesorting/imported.py

```diff
@@ -58,3 +67,74 @@ def get_sorting(cls, key):
         raise NotImplementedError(
             "Imported spike sorting does not have a `get_sorting` method"
         )
+
+    def add_annotation(
+        self, key, id, label=[], annotations={}, merge_annotations=False
+    ):
+        """Manually add annotations to the spike sorting output
+
+        Parameters
+        ----------
+        key : dict
+            restriction key for ImportedSpikeSorting
+        id : int
+            unit id
+        label : List[str], optional
+            list of str labels for the unit, by default None
+        annotations : _type_, optional
+            dictionary of other annotation values for unit, by default None
+        merge_annotations : bool, optional
+            whether to merge with existing annotations, by default False
+        """
+        if not len(self & key) == 1:
+            raise ValueError("ImportedSpikeSorting key must be unique")
+        unit_key = {**key, "id": id}
+        if ImportedSpikeSorting.Annotations & unit_key:
+            if not merge_annotations:
+                raise ValueError("Unit already has annotations")
+            existing_annotations = (
+                ImportedSpikeSorting.Annotations & unit_key
+            ).fetch(as_dict=True)[0]
+            existing_annotations["label"] = (
+                existing_annotations["label"] + label
+            )
```

Comment:
`fetch(as_dict=True)[0]` will give the same result as `fetch1()`, but hide a possible error state in which there is not exactly one result. Do you want to only edit the first of multiple valid entries, or should we only expect one?


```suggestion
        annotation_query = ImportedSpikeSorting.Annotations & unit_key
        if annotation_query and not merge_annotations:
            raise ValueError(f"Unit already has annotations: {annotation_query}")
        elif annotation_query:
            existing_annotations = annotation_query.fetch1()
            existing_annotations["label"] += label
```